### PR TITLE
[CELEBORN-229][FEATURE] Support collect metrics with customized labels

### PR DESCRIPTION
**Pull request by nafiyAix**
### What changes were proposed in this pull request?
Add labels field for Celeborn metrics

### Why are the changes needed?
Support add customized labels when intializing metrics, thus we can add more fine-grained monitoring on 3rd-party monitoring system.

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
Unit test

### CODE REVIEW

File: common/src/main/scala/org/apache/celeborn/common/metrics/source/AbstractSource.scala [hunk 1]
```diff
@@ -60,29 +67,32 @@ abstract class AbstractSource(conf: CelebornConf, role: String)
...
-  def addGauge[T](name: String, f: Unit => T): Unit = {
+  def addGauge[T](
+      name: String,
+      f: Unit => T,
+      labels: Map[String, String] = Map.empty[String, String]): Unit = {
     val supplier: MetricRegistry.MetricSupplier[Gauge[_]] = new GaugeSupplier[T](f)
     val gauge = metricRegistry.gauge(name, supplier)
-    namedGauges.add(NamedGauge(name, gauge))
+    namedGauges.add(NamedGauge(name, gauge, labels))
   }
```
Comment (AngersZhuuuu) [inline common/src/main/scala/org/apache/celeborn/common/metrics/source/AbstractSource.scala:77]: Maybe we can support  both
```
def addGauge[T](name: String, f: Unit => T): Unit = { .... }
```
and 

```
 def addGauge[T](
      name: String,
      f: Unit => T,
      labels: Map[String, String]): Unit = {
   ......
}
```

Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/metrics/source/AbstractSource.scala:77]: namedGauges.add(NamedGauge(name, gauge, labels + "role" -> role))

File: common/src/main/scala/org/apache/celeborn/common/metrics/source/AbstractSource.scala [hunk 2]
```diff
@@ -300,7 +314,14 @@ abstract class AbstractSource(conf: CelebornConf, role: String)
...
 
-  val label = s"""{role="$role"}"""
+  def getLabels(metricLabels: Map[String, String]): String = {
+    val sb = new StringBuilder()
+    sb.append("{")
+    metricLabels.foldLeft(sb.append(s"""role="$role""""))((_, entry) =>
+      sb.append(s""" ${entry._1}="${entry._2}""""))
+    sb.append("}")
+    sb.toString()
```
Comment (AngersZhuuuu) [inline common/src/main/scala/org/apache/celeborn/common/metrics/source/AbstractSource.scala:323]:
```
metricLabels.map { case (key, value) => s""${key}="${value}"""" }.mkString(" ")
```

File: common/src/main/scala/org/apache/celeborn/common/metrics/source/AbstractSource.scala [hunk 3]
```diff
@@ -30,13 +30,20 @@ import org.apache.celeborn.common.internal.Logging
 import org.apache.celeborn.common.metrics.{ResettableSlidingWindowReservoir, RssHistogram, RssTimer}
 import org.apache.celeborn.common.util.{ThreadUtils, Utils}
 
-case class NamedCounter(name: String, counter: Counter)
+private[source] trait MetricLabels {
+  val labels: Map[String, String]
```
Comment (AngersZhuuuu) [inline common/src/main/scala/org/apache/celeborn/common/metrics/source/AbstractSource.scala:34]:
```
def labelString = xxx
```

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**